data-structures.md

```diff
@@ -56,17 +81,25 @@ type Block struct {
 
 	// Height is the chain height of this block.
 	Height Uint64
-    
-    // StateRoot is a cid pointer to the state tree after application of the
+
+	// StateRoot is a cid pointer to the state tree after application of the
 	// transactions state transitions.
 	StateRoot Cid
 
-	// Messages is the set of messages included in this block
+	// ECDSAMessages is the set of messages signed with ECDSA included in this block
 	// TODO: should be a merkletree-ish thing
-	Messages []SignedMessage
+	ECDSAMessages []SignedMessage
+  
+  // BLSSignatureAggregate is the SignedMessage contianing the aggreate signature of 
+  // all messages signed with the BLS signature scheme.
+  BLSSignatureAggregate SignedMessage
+  
+  // BLSMessages is the set of messages with associated BLS signatures contained
+  // in BLSSignatureAggregate.
+  BLSMessages []Message
 
 	// MessageReceipts is a set of receipts matching to the sending of the `Messages`.
-    // TODO: should be the same type of merkletree-list thing that the messages are
+	// TODO: should be the same type of merkletree-list thing that the messages are
 	MessageReceipts []MessageReceipt
```

Comment:
@whyrusleeping: what are `MessageReceipts`?

Reply:
They are defined below and store the result of applying the message

Reply:
I see the `struct` definition. Do we have any exposition as to why we need them? Given that they are an on-chain quantity, I would claim that it's important to declare exactly why we have them. This would be the best place to put that explanation.